compiler/rustc_session/src/session.rs

```diff
@@ -738,6 +792,8 @@ impl Session {
         let found_negative = requested_features.clone().any(|r| r == "-crt-static");
         let found_positive = requested_features.clone().any(|r| r == "+crt-static");
 
+        // JUSTIFICATION: necessary use of crate_types directly
```

Comment:
why is that necessary?

Reply:
It is called from `validate_commandline_args_with_session_available` at which point `sess.crate_types` is not yet initialized. There is a FIXME a couple lines below about this.

Reply:
I've modified the comment to reference the FIXME below.